emily/handler/src/api/handlers/withdrawal.rs

```diff
@@ -336,71 +335,121 @@ pub async fn update_withdrawals(
         let api_state = accessors::get_api_state(&context).await?;
         api_state.error_if_reorganizing()?;
 
-        let is_trusted_key = context.settings.trusted_reorg_api_key == api_key;
         // Signers are only allowed to update withdrawals to the accepted state.
-        if !is_trusted_key {
-            let is_unauthorized = body
-                .withdrawals
-                .iter()
-                .any(|withdrawal| withdrawal.status != Status::Accepted);
+        let is_allowed = !body
+            .withdrawals
+            .iter()
+            .any(|withdrawal| withdrawal.status != Status::Accepted);
```

Comment:
I think this would be slightly easier to read:
```rust
let is_allowed = body
    .withdrawals
    .iter()
    .all(|withdrawal| withdrawal.status == Status::Accepted);
```
Also, an empty withdrawals object wouldn't return a `401 Forbidden` with the proposed change. I'm also in favor of returning early and logging if we get an empty withdrawals object.

Reply:
[5f335df](https://github.com/stacks-sbtc/sbtc/pull/1586/commits/5f335dfc033b2fb2ee1b1e40d91010a462bde2d7) simplified reading here

Reply:
About changing return codes I'd say it is out of scope for this pr

Reply:
Do you mean returning early is out of scope? I'm not pushing for the early return, it's likely not an issue either way. And I didn't want to change the return code here, just avoid sending the wrong one when the withdrawals object is empty. With https://github.com/stacks-sbtc/sbtc/pull/1586/commits/5f335dfc033b2fb2ee1b1e40d91010a462bde2d7 we'll avoid that so we're good.